code/modules/religion/religion_structures.dm

```diff
@@ -10,37 +10,16 @@
 	pass_flags = LETPASSTHROW
 	can_buckle = TRUE
 	buckle_lying = 90 //we turn to you!
-	var/datum/religion_sect/sect_to_altar // easy access!
-	var/datum/religion_rites/performing_rite
+	///Avoids having to check global everytime by referencing it locally.
+	var/datum/religion_sect/sect_to_altar
 
-/obj/structure/altar_of_gods/examine(mob/user)
+/obj/structure/altar_of_gods/Initialize(mapload)
 	. = ..()
-	var/can_i_see = FALSE
-	if(isobserver(user))
-		can_i_see = TRUE
-	else if(isliving(user))
-		var/mob/living/L = user
-		if(L.mind?.holy_role)
-			can_i_see = TRUE
-
-	if(!can_i_see || !sect_to_altar)
-		return
-
-	. += "<span class='notice'>The sect currently has [round(sect_to_altar.favor)] favor with [GLOB.deity].</span>"
-	if(!sect_to_altar.rites_list)
-		return
-	. += "List of available Rites:"
-	. += sect_to_altar.rites_list
+	reflect_sect_in_icons()
 
-
-/obj/structure/altar_of_gods/Initialize(mapload)
+/obj/structure/altar_of_gods/ComponentInitialize()
 	. = ..()
-	if(GLOB.religious_sect)
-		sect_to_altar = GLOB.religious_sect
-		if(sect_to_altar.altar_icon)
-			icon = sect_to_altar.altar_icon
-		if(sect_to_altar.altar_icon_state)
-			icon_state = sect_to_altar.altar_icon_state
+	AddComponent(/datum/component/religious_tool, ALL, FALSE, CALLBACK(src, .proc/reflect_sect_in_icons), null)
```

Comment:
```suggestion
	AddComponent(/datum/component/religious_tool, ALL, FALSE, CALLBACK(src, .proc/reflect_sect_in_icons))
```
`null` can't be passed as an argument, it's the same as not passing anything. This is intentional design in DM.

Reply:
bruh CRINGE